com.unity.ml-agents/Runtime/InferenceBrain/ApplierImpl.cs

```diff
@@ -13,18 +13,26 @@ namespace MLAgents.InferenceBrain
     /// </summary>
     public class ContinuousActionOutputApplier : TensorApplier.IApplier
     {
-        public void Apply(TensorProxy tensorProxy, IEnumerable<AgentIdActionPair> actions)
+        public void Apply(TensorProxy tensorProxy, IEnumerable<int> actionIds, Dictionary<int, float[]> lastActions)
         {
             var actionSize = tensorProxy.shape[tensorProxy.shape.Length - 1];
             var agentIndex = 0;
-            foreach (var idActionPair in actions)
+            foreach (int agentId in actionIds)
             {
-                var actionValue = new float[actionSize];
-                for (var j = 0; j < actionSize; j++)
+                if (lastActions.ContainsKey(agentId))
                 {
-                    actionValue[j] = tensorProxy.data[agentIndex, j];
+                    var actionValue = lastActions[agentId];
+                    if (actionValue == null)
```

Comment:
Are you confident that `actionValue != null` implies `actionValue.Length == actionSize`? I guess as long as the agent can't change its action space and ids can't get reused it should be OK...